src/test/java/test/sanitycheck/CheckSuiteNamesTest.java

```diff
@@ -81,4 +88,26 @@ public void checkXmlSuiteAddition() throws ParserConfigurationException, SAXExce
     tng.setXmlSuites(parser.parseToList());
     tng.initializeSuitesAndJarFile();		
   }
+  
+  @Test(description = "Verify that same suite is not added multiple times")
+  public void validateDuplicateSuiteAddition() throws ParserConfigurationException, SAXException, IOException
```

Comment:
Could you add a test for #829 too? We should check that it is still possible to add the same child suite many times.